### PR TITLE
fix: Fix off by one error when retrying id registry event on sync

### PR DESCRIPTION
**Pull request by sanjayprabhu**
## Motivation

Fixes https://github.com/farcasterxyz/hub-monorepo/issues/892

And helps with https://github.com/farcasterxyz/hub-monorepo/issues/898

## Change Summary

There's an off by one error when we retry fetching a id registry event which meant we weren't actually querying the events. This fixes the issue, and adds a flag to trigger a manual resync if desired

## Merge Checklist

_Choose all relevant options below by adding an `x` now or at any time before submitting for review_

- [x] PR title adheres to the [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/) standard
- [x] PR has a [changeset](../CONTRIBUTING.md#35-adding-changesets)
- [x] PR has been tagged with a change label(s) (i.e. documentation, feature, bugfix, or chore)
- [x] PR does not require changes to the [protocol](https://github.com/farcasterxyz/protocol)
- [x] All [commits have been signed](../CONTRIBUTING.md#22-signing-commits)

## Additional Context

If this is a relatively large or complex change, provide more details here that will help reviewers

### CODE REVIEW

File: apps/hubble/src/cli.ts [hunk 1]
```diff
@@ -78,6 +78,7 @@ app
   .option('--admin-server-host <host>', "The host the admin server should listen on. (default: '127.0.0.1')")
   .option('--db-name <name>', 'The name of the RocksDB instance')
   .option('--rebuild-sync-trie', 'Rebuilds the sync trie before starting')
+  .option('--resync-events', 'Resyncs events from the Farcaster contracts before starting')
```
Comment (pfletcherhill) [inline apps/hubble/src/cli.ts:81]: Can we call this `resync-eth-events` just to draw more separation from hub events?

File: apps/hubble/src/eth/ethEventsProvider.ts [hunk 2]
```diff
@@ -146,9 +146,9 @@ export class EthEventsProvider {
     return this._lastBlockNumber;
   }
 
-  public async start() {
+  public async start(resyncEvents?: boolean) {
```
Comment (pfletcherhill) [inline apps/hubble/src/eth/ethEventsProvider.ts:149]: We already pass in other config options to the constructor or the static `build` method. Can we pass this option there instead?

**Review by pfletcherhill — COMMENTED** [review]
Two small comments

File: apps/hubble/src/cli.ts (hunk 1 above)
Comment (sanjayprabhu) [reply apps/hubble/src/cli.ts:81]: Addressed in https://github.com/farcasterxyz/hub-monorepo/pull/916

**Pull request merged**